cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/ShuntConversion.java

```diff
@@ -70,12 +63,94 @@ public void convert() {
             throw new IllegalStateException("Unexpected shunt type: " + shuntType);
         }
         identify(adder);
-        connect(adder);
+        connectWithOnlyEq(adder);
         ShuntCompensator shunt = adder.add();
         addAliasesAndProperties(shunt);
 
-        PowerFlow f = powerFlowSV();
-        context.convertedTerminal(terminalId(), shunt.getTerminal(), 1, f);
+        convertedTerminalsWithOnlyEq(shunt.getTerminal());
         context.regulatingControlMapping().forShuntCompensators().add(shunt.getId(), p);
+        shunt.setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.NORMAL_SECTIONS, String.valueOf(normalSections));
+    }
+
+    public static void update(ShuntCompensator shuntCompensator, PropertyBag cgmesData, Context context) {
+        updateTerminals(shuntCompensator, context, shuntCompensator.getTerminal());
+        updateSections(shuntCompensator, cgmesData, context);
+
+        boolean controlEnabled = cgmesData.asBoolean(CgmesNames.CONTROL_ENABLED, false);
+        updateRegulatingControl(shuntCompensator, controlEnabled, context);
+    }
+
+    private static void updateSections(ShuntCompensator shuntCompensator, PropertyBag cgmesData, Context context) {
+        int normalSections = Integer.parseInt(shuntCompensator.getProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.NORMAL_SECTIONS));
```

Comment:
This line will throw an exception if the property is absent. You should use a default value when `getProperty(...)` returns `null`.

Reply:
Fixed